frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -1260,6 +1260,24 @@ export class ExpressionDimension extends Dimension {
       semantic_type = base_type;
     }
 
+    // FIX
+    // when the app uses dimension().field().dimension().field()
+    // the last generated field is badly formatted
+    const resultMetadata = query?.question()?.getResultMetadata?.();
```

Comment:
When the app uses `dimension().field().dimension().field()`, the last generated field is badly formatted so we use the `metadata` to retrieve the valid property.